From c7011e1dda9a42646864b61acd76ee129d4deba6 Mon Sep 17 00:00:00 2001
From: motten <motten@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 17 Jan 2017 14:56:50 +0000
Subject: [PATCH 0740/1000] MANTIS 0032431 \version 1.0.1.20 \li 17.01.2017 \li
 MOt - Mantis 0032431: ScheduledTask \li set range for the delay in the
 constructor of ScheduledTask [0, 12*60]

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1139 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Tools/inc/local_def_msystools.h    | 11 ++++++++---
 .../Massai/cpp/Tools/src/ScheduledTask.cpp        | 15 ++++++++++++---
 2 files changed, 20 insertions(+), 6 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h b/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h
index 46566571..f609d9d1 100644
--- a/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h
+++ b/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h
@@ -1,5 +1,5 @@
-#define FILE_VER_BIN     1,0,1,19
-#define FILE_VER_STR     "FileVersion", "1.0.1.19"
+#define FILE_VER_BIN     1,0,1,20
+#define FILE_VER_STR     "FileVersion", "1.0.1.20"
 
 /*!
 \file
@@ -7,12 +7,17 @@
 
 \page massai_tools.lib Release Notes - massai_tools.lib
 
-@version 1.0.1.19
+@version 1.0.1.20
 
 \section changehistory Change history
 
 tools.lib
 
+\version 1.0.1.20
+\li 17.01.2017
+\li MOt - Mantis 0032431: ScheduledTask
+\li set range for the delay in the constructor of ScheduledTask [0, 12*60]
+
 \version 1.0.1.19
 \li 01.12.2016
 \li SteT - Mantis  0043868: integration of EZ-Access device for KIS kiosk
diff --git a/SelfServiceCommon/Massai/cpp/Tools/src/ScheduledTask.cpp b/SelfServiceCommon/Massai/cpp/Tools/src/ScheduledTask.cpp
index 16393d4e..7490dc1e 100644
--- a/SelfServiceCommon/Massai/cpp/Tools/src/ScheduledTask.cpp
+++ b/SelfServiceCommon/Massai/cpp/Tools/src/ScheduledTask.cpp
@@ -169,15 +169,24 @@ namespace MASSAI
       {
         if (delay != 0)
         {
+          //Maximum for the delay is 12 hours 
+          if(delay > 12*60) 
+          {
+            TRACE(MajorWarning_HERE, "WARNING: Delay is out of the valid range of [0, 12*60] '0 minutes to 12 hours', default set to the maximum!");
+
+            delay = 12 * 60;
+          }         
+          
           randomDelay = rand() % delay + 1;
+
+          TRACE(HERE, "Calculated random delay [%ld]", randomDelay);
         }
       }
       else
       {
-        stringstream execeptionInfo("");
-        execeptionInfo << "Delay is out of the valid range! Delay=" << delay;
+        TRACE(MajorWarning_HERE, "WARNING: Delay is out of the valid range of [0, 12*60] '0 minutes to 12 hours', default set to the minium!");
 
-        throw std::invalid_argument(execeptionInfo.str().c_str());
+        TRACE(HERE, "Random delay is set to [%ld]", randomDelay);
       }
       
       unsigned long minutes_of_execution_time = hour*60 + minutes + randomDelay;
-- 
2.41.0.windows.1

